entry.go

```diff
@@ -20,7 +20,48 @@
 
 package zap
 
-import "time"
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+var _cePool = sync.Pool{
+	New: func() interface{} {
+		return &CheckedEntry{}
```

Comment:
Sure, but my comment was really asking: we should benchmark and see if its worth it to special case space for 1, 2, or 3 facilities out side of the slice so that we can support arbitrary. I.E. using a static `[2]Facility` plus an `int` field (or nil checks) to use it before the slice.